src/events/pollVotationResults.js

```diff
@@ -6,15 +6,28 @@ module.exports = {
   async execute(message) {
     const { embeds, author, client, channelId } = message;
 
-    if (author.bot && embeds[0]?.data?.type !== 'poll_result') {
+    if (
+      author.bot &&
+      Array.isArray(embeds[0]?.fields) &&
+      embeds[0]?.data?.type !== 'poll_result'
+    ) {
       return;
     }
 
     const userMentionField = embeds[0]?.fields.find(
       (field) => field.name === 'poll_question_text'
     );
+
+    if (userMentionField === undefined) {
+      return;
+    }
+
     const parts = userMentionField?.value.split('|').map((part) => part.trim());
-    const userMentioned = `<@${parts[1]}>`;
+    if (!Array.isArray(parts)) {
+      return;
+    }
+
+    const userMentioned = parts?.[1] ? `<@${parts[1]}>` : null;
```

Comment:
We could use the same early return to confirm the line 30 will work, so in line 30 we can avoid the validation
```suggestion
    if (!Array.isArray(parts) || !parts[1]) {
      return;
    }

    const userMentioned = `<@${parts[1]}>`;
```

Reply:
Completed